### ISSUE TITLE
Possible false-positive with @throws tag in closure

### ISSUE THREAD

**Issue by sjdaws** [kalessil/phpinspectionsea#686]
Hi,

If I have a method with a closure I am unable to annotate it in a way that satisfies both PHP inspections EA and PhpStorm.

Code block:
```
/**
 * Register something in the container
 *
 * @return void
 */
public function register(): void
{
    $instance = function () {
        $test = new Test($this->app->make(Request::class));

        // Validate, will throw an exception if there is an error
        $test->validate();

        return $test;
    };

    $this->app->bind(Test::class, $instance);
}
```

In this case inspections reports `validate` throws an exception, which is correct:
![screen shot 2017-12-13 at 10 15 51 am](https://user-images.githubusercontent.com/659316/33913631-bc730e6e-dfee-11e7-82f6-3fb0b4049708.png)

If I delcare exception via `@throws` to the parent method PhpStorm complains that the exception is never thrown:
![screen shot 2017-12-13 at 10 16 09 am](https://user-images.githubusercontent.com/659316/33913645-ca1ff2de-dfee-11e7-9422-071512917666.png)

The helper menu automatically generates a `@throws` for the closure, but this doesn't help:
![screen shot 2017-12-13 at 10 19 27 am](https://user-images.githubusercontent.com/659316/33913734-2d80ab2a-dfef-11e7-9d3d-ea8770859818.png)
![screen shot 2017-12-13 at 10 21 10 am](https://user-images.githubusercontent.com/659316/33913787-62ef91f4-dfef-11e7-9c51-70f367859de1.png)

Should inspections EA be reading the annotation from the closure rather than the parent method?

**Comment by kalessil** [comment]
It should, thank you for reporting @sjdaws

**Comment by kalessil** [comment]
@sjdaws : I've rechecked the case and seems it's correctly reported. Once JetBrains team delivers https://youtrack.jetbrains.com/issue/WI-40818 the PhpDoc warning will go away.

Sorry for such news.
